File: c-sharp/SearchInsertPosition.cs
Console.WriteLine(new Solution().SearchInsert(new int[] { 1, 3, 5, 6 }, 2));

public class Solution
{
    public int SearchInsert(int[] nums, int target)
    {
        if (nums.Length == 0) return 0;

        if (nums.IndexOf(target) != -1)
        {
            return nums.IndexOf(target);
        }
        else
        {
            // If the target is greater than the last elements value, it should be inserted at the end.
            if (target > nums[nums.Length - 1])
            {
                return nums.Length;
            }
            // Else if its smaller than the first element, it should be inserted at the beginning.
            else if (target < nums[0])
            {
                return 0;
            }
            // Value wasm't found so iterate over the array for the insertion point
            // Check if the value is between two numbers
            else
            {
                int insertPoint = 0;
                for (int i = 0; i < nums.Length; i++)
                {
                    if (i >= 0 && i < nums.Length - 1)
                    {
                        if (target > nums[i] && target < nums[i + 1])
                        {
                            insertPoint = i + 1;
                        }
                    }
                }
                return insertPoint;
            }
        }
    }
}

File: python/search_insert_position.py
class Solution:
    def searchInsert(self, nums: list[int], target: int) -> int:
        if len(nums) == 0:
            return 0

        if target in nums:
            return nums.index(target)
        else:
            if target > nums[len(nums) - 1]:
                return len(nums)
            elif target < nums[0]:
                return 0
            else:
                insertPoint = 0

                for i in range(len(nums)):
                    if i >= 0 and i < len(nums) - 1:
                        if target > nums[i] and target < nums[i + 1]:
                            insertPoint = i + 1
                            break

                return insertPoint


s = Solution()
print(s.searchInsert([1, 3, 5, 6], 2))
